website/docs/guides/2.0-upgrade-guide.html.markdown

```diff
@@ -0,0 +1,348 @@
+---
+layout: "azurerm"
+page_title: "Azure Resource Manager: 2.0 Upgrade Guide"
+sidebar_current: "docs-azurerm-guide-2.0-upgrade"
+description: | -
+    This page documents how to migrate between two resources in the Azure Provider which have been renamed.
+
+---
+
+# v2.0 of the AzureRM Provider
+
+Terraform initially shipped support for the AzureRM Provider back in December 2015.
+
+Since then we've added support for 191 Resources, 58 Data Sources and have launched a couple of related Providers in the form of [the Azure Active Directory Provider](https://www.terraform.io/docs/providers/azuread/index.html) and [the Azure Stack Provider](https://www.terraform.io/docs/providers/azurestack/index.html).
+
+Version 2.0 of the AzureRM Provider is a major release and as such includes some larger-scale changes which are outlined in this document.
+
+-> **NOTE:** This guide is a Work In Progress and as such fields may be added to this guide until version 2.0 of the AzureRM Provider is released.
+
+At a high level, we're intending for version 2.0 to include the following changes:
+
+* Existing Resources will be required to be imported
+* Custom Timeouts will be available on Resources - this will allow you to specify a custom timeout for provisioning the resource in your Terraform Configuration [using the `timeouts` block](https://www.terraform.io/docs/configuration/resources.html#timeouts).
+* New resources for Virtual Machines and Virtual Machine Scale Sets
+* Removing Fields, Data Sources and Resources which have been deprecated
+
+Each of these topics is covered in more detail below, however please note that this guide is a Work In Progress until version 2.0 of the AzureRM Provider is released and thus things may be added/changed as necessary.
+
+## Requiring Resources to be Imported
+
+Terraform allows for existing resources which have been created outside of Terraform to be Imported into Terraform's State. Once a resource is imported into the state, it's possible for Terraform to track changes and manage this resource. The Azure Provider allows Importing existing resources into the state (using `terraform import`) for (almost) every resource.
+
+Version 2.0 of the Azure Provider aims to solve an issue where it's possible to unintentionally import resources into the state by running `terraform apply`. To explain this further, the majority of Azure's API's are Upserts - which means that a resource will be updated if it exists, otherwise it'll be created.
+
+Where the unique identifier for (most) Azure resources is the `name` (rather than for example an `aws_instance` where AWS will generate a different unique identifier) - it's possible that users may have unintentionally imported resources into Terraform when running `terraform apply` on an existing resource.
```

Comment:
Should we change this to `azurerm_policy?
```suggestion
Where the unique identifier for (most) Azure resources is the `name` (rather than for example an `azurerm_policy_definition` where the provider will generate a different unique identifier) - it's possible that users may have unintentionally imported resources into Terraform when running `terraform apply` on an existing resource.
```

Reply:
I intentionally picked AWS here since almost(?) all resources have a unique identifier (whereas they're the exception in Azure, rather than the rule) - so I think this is worth leaving?